src/app/loadout/ingame/ingame-loadout-utils.ts

```diff
@@ -36,3 +39,68 @@ export function useItemsFromInGameLoadout(loadout: InGameLoadout) {
     [loadout.items, allItems]
   );
 }
+
+/**
+ * does this game loadout, meet the requirements of this DIM loadout:
+ *
+ * does it include the items the DIM loadout would equip,
+ * and represent a full application of the DIM loadout's required mods?
+ */
+export function implementsDimLoadout(
+  inGameLoadout: InGameLoadout,
+  dimLoadout: Loadout,
+  defs: D2ManifestDefinitions
+) {
```

Comment:
This function should probably be called with a fully resolved loadout (including resolved loadout mods) 

1. DIM loadouts have item migration, e.g. for crafted weapons, where the IDs don't match and instead the crafted timestamp is used.
2. We just merged some loadout mod migration in #9287, so some old loadouts now have old stat mod hashes but resolve to new ones. `getModsFromLoadout` takes care of that.
3. DIM loadouts always store the regular version of a mod and apply what's available, while an in-game loadout may store the reduced-cost copy. This function should probably involve https://github.com/DestinyItemManager/DIM/blob/73f2f5e705491f09b6d3910f830bb4184047e931/src/app/loadout/mod-utils.ts#L128-L133 when checking mods.

Reply:
To clarify, for the items you probably don't need a ResolvedLoadoutItem, just need to call `findItemForLoadout` to find the correct item.

Reply:
ok this now uses resolved mods and items. hopefully game-side loadout plays nice with mod discounts too. stuff is accurately detecting though, in testing here.

Reply:
still need to screw around with activating/deactivating artifact discounts but so far matching remarkably well using the mod resolver.